### DISCUSSION TITLE
LuaSnip doesn't complete Java package name

### DISCUSSION THREAD

**Discussion by VisionaryAppDev** [L3MON4D3/LuaSnip#836] [Q&A]
Just couldn't get LuaSnip to complete java package name and I am not sure if it's the default behavior of the template. When completing class name, it just work pretty well.

https://user-images.githubusercontent.com/27724011/226945692-0f4969dc-dca1-4d75-9992-bcf3f4605e56.mp4

**Comment by evesdropper** [comment]
The snippet docstring gives it away - when you show the snippet for `package`, the insert is `${1:PackageName}`, so it inserts text `PackageName` regardless of the package name; whereas for the class snippet, the insert is `${TM_FILENAME_BASE}` - hence it's "smart" in picking up the filename.

**Reply by VisionaryAppDev** [reply]
Currently I am using "friendly-snippets" from rafamadriz. So, is there anyway to achieve the goal then?

**Reply by evesdropper** [reply]
Yeah sure, you can write a snippet in Lua. I'm assuming you don't have any custom snippets yet, so you'd need to add the Lua Loader:

```lua
require("luasnip.loaders.from_lua").load({ paths = "SNIPPETDIR })
```

and populate new Java snippets in `java.lua`.

Here's what I came up with:

```lua
s({ trig='package', name='package', dscr='package'},
    fmta([[
    package <>
    ]],
    { f(function (_, _)
        pwdstring = vim.api.nvim_call_function('fnamemodify', {vim.loop.cwd(), ":t"})
    return pwdstring
    end) }
    )),
```
